lua/neo-tree/sources/common/commands.lua

```diff
@@ -342,6 +342,25 @@ M.delete_visual = function(state, selected_nodes, callback)
   fs_actions.delete_nodes(paths_to_delete, callback)
 end
 
+M.trash = function(state, callback)
+  local tree = state.tree
+  local node = tree:get_node()
+  if node.type == "message" then
+    return
+  end
+  fs_actions.trash_node(node.path, callback)
+end
+
+M.trash_visual = function(state, selected_nodes, callback)
+  local paths_to_delete = {}
+  for _, node_to_delete in pairs(selected_nodes) do
```

Comment:
I think the only problem here is that it should be **i**pairs, it is a list and not a key/value map.

```suggestion
  for _, node_to_delete in ipairs(selected_nodes) do
```